### ISSUE TITLE
anova.merMod P-value for zero df

### ISSUE THREAD

**Issue by MetaEntropy** [lme4/lme4#583]
# Short description
Comparing two equivalent [g]lmer fits (same degrees of freedom) with anova.merMod yields unstable P-values, that oscillate between 1 and 0.
anova(model1, model2) can even provide a P-value equal to 1 while anova(model1, model2) provide a P-value equal to 0.

# Example 1 (optimizer)
```R
expo=log(1:1000)
group=rep(c(0,1),500)
outcome=1:1000
# Use different optimizers to induce minor floating point errors
mod1=lmer(outcome ~ expo+(expo|group), REML=FALSE, control=lmerControl(optimizer="Nelder_Mead"))
mod2=lmer(outcome ~ expo+(expo|group), REML=FALSE, control=lmerControl(optimizer="nloptwrap"))
logLik(mod1)-logLik(mod2) # Slightly positive due to calculation error (e.g. 9.004361e-07)
anova(mod1, mod2)[["Pr(>Chisq)"]][2] # P-value equal to 1
anova(mod2, mod1)[["Pr(>Chisq)"]][2] # P-value equal to 0
```
# Example 2 (offset)
```R
expo=log(1:1000)
group=rep(c(0,1),500)
outcome=1:1000
# Use different offsets to induce minor floating point errors
off1=rep(1, 1000)
mod1=lmer(outcome ~ expo+(expo|group)+offset(off1), REML=FALSE)
mod2=lmer(outcome ~ expo+(expo|group), REML=FALSE)
logLik(mod1)-logLik(mod2) # Slightly positive due to calculation error (e.g. 3.489949e-07)
anova(mod1, mod2)[["Pr(>Chisq)"]][2] # P-value equal to 1
anova(mod2, mod1)[["Pr(>Chisq)"]][2] # P-value equal to 0

```
# Cause
The difference of deviance is almost zero but the chi-square distribution has zero degrees of freedom, yeilding very different values when the actual difference of deviance oscillates around zero.
`pchisq(-1e-15, df=0, lower.tail=FALSE)` is 1
`pchisq(1e-15, df=0, lower.tail=FALSE)` is 0
`pchisq(0, df=0, lower.tail=FALSE)` is 1

# Expected behavior?
Comparing two strictly equivalent models do not make much sense.
anova.glm and anova.lm return NA as P-value in that case!

Suggested solution: adopt the same behavior as anova.glm and anova.lm, i.e. yield NA P-value when the difference of degrees of freedom is zero.

**Comment by bbolker** [comment]
This sounds eminently sensible and not hard to implement.  Thanks!

**Comment by bbolker** [comment]
The fix for this accidentally got rolled into https://github.com/lme4/lme4/pull/587 , but should be merged shortly ...

**Comment by bbolker** [comment]
Fixed in 4949036c
